README-development.md

```diff
@@ -1,3 +1,49 @@
+# Operating System
+
+## Prerequisites
+
+### Usually packaged by operating systems
+- git
+- ostree
+- rpm-ostree
+- qemu-img
+- fedpkg
+- mock
+
+### May need to be built from source
+- [coreos-assembler](https://github.com/cgwalters/coreos-assembler/): [container](http://quay.io/cgwalters/coreos-assembler)
+- [rpmdistro-gitoverlay](https://github.com/projectatomic/rpmdistro-gitoverlay)
+- [imagefactory-plugins-TinMan](https://github.com/redhat-imaging/imagefactory)
+
+## Building
+
+- Clone ``openshift/os``
+- Move into the cloned repo
+- Build packages from source repos:
+ - ``rpmdistro-gitoverlay init``
+ - ``rpmdistro-gitoverlay resolve --fetch-all``
+ - ``rpmdistro-gitoverlay build``
+- Make the ostree
+ - ``export REPO=/tmp/repo``
+ - ``ostree init --repo=$REPO --mode=archive``
+ - ``podman run --privileged --rm -v $REPO:/srv quay.io/cgwalters/coreos-assembler``
+ - ``cd /srv && coreos-assembler --repo=/srv --cachedir=cache host.yml``
```

Comment:
I used the `coreos-assembler` container to run all the commands up to here (including the rdgo ones).  

If you follow the steps verbatim, the `coreos-assembler` command will fall over because there is no `openshift.repo` file available in `openshift/os`

I was able to generate the `openshift.repo` with the `generate-openshift-repo` command in the `openshift/os` repo, but then we fall over due to missing the Red Hat GPG key.

So I propose that the additional steps of:

1) running `generate-openshift-repo`
2) copying the Red Hat GPG key into `/etc/pki/rpm-gpg/`

...then running the `coreos-assembler` command.

(It might also be worth noting separately that the `coreos-assembler` command is not going to work unless you have access to the Red Hat stage CDN, just to dissuade anyone who is curious)

Reply:
I was also going through the steps, and hit the same. For the former there seems to be already a `make repo-refresh` (similarly for init). For the latter I'm confused because it seems to bail on a `gpgcheck=0` repo; is there a way to hook repo configs to relative paths from this git-repo?

Reply:
Yeah, we can.